lib/pbench/server/cache_manager.py

```diff
@@ -93,16 +93,19 @@ def __init__(self, path: Path):
         """
         self.name = path.name
         self.location = path
+        self.resolve_path = path.resolve(strict=True)
```

Comment:
First, a quibble:  I think we need to keep in mind that symlinks with relative referents are likely to be a very different beast from symlinks with absolute referents.  I realize, @dbutenhof, that your example was meant to be "quick", but a quick perusal of the tarballs in the repo show no "archive members" which have absolute pathnames and all members which are symlinks have relative referents.  (Of course, this doesn't mean that someone _couldn't_ hand us one with absolute paths in it, but it's a starting point.)

But, ignoring the leading slash thing, entries in the cachemap for each of `foo` and `dir/foo` will have their `resolve_path` attributes set to the final resolution...which, for the record, is an absolute path (to wit, `<absolute_prefix>/dir1/foo`).

And, I think that this is mostly OK.  I depends on how we intend to use this information, and, for the immediate future, (I believe...) we don't intend to do anything with it other than display it, so it's all good.  But, having deliberation and documentation would be good....

(But, I'm not sure that we want the `strict=True` here...at least, not without some error handling to deal with broken symlinks.)

Reply:
I think we never want to expand symlinks in the cache map, because the expansion will be an absolute path leading through our temporary unpack directory, and we don't want to be always unpacking the entire tarball to access a file. What we need to know is the *relative path* within the tarball and the *absolute* path of the (temporarily) cached file. When we get to sharing the map across independent component instances, the extracted cache (to the extent that we use local files at all) would be potentially different for each. (Though if we cache in Redis, the "actual path" will just be an object ID.)

So if the "actual path" for this cache map is going to be the file at the end of the link chain, we really do need to `readlink` and keep all our paths relative to the base of the tarball.

Reply:
> I think we never want to expand symlinks in the cache map, because the expansion will be an absolute path leading through our temporary unpack directory, and we don't want to be always unpacking the entire tarball to access a file.

@dbutenhof, I'm not sure what you mean by "expand symlinks", but the result does not _have_ to be an absolute path (although, that's what we're getting at the moment from `Path.resolve()`, if it doesn't raise an exception), and we don't _have_ to re-unpack the tarball to find the referent.  (On the other hand, we probably need to be much more careful when we do unpack the tarball, to defend against malicious contents....)

If and/or when we do provide support for referencing files and/or directories via symlinks (which I think we should _not_ do in this PR...), then we'll have to take on the issue of figuring out what the symlink refers to.  (E.g., in the case of file referents, a symlink might contain a relative path or an absolute path and that path might refer to a file inside the results or outside the results; the same is true in the case of directories, but we also have to worry about whether the referent is a parent directory of the current location.)

> What we need to know is the _relative path_ within the tarball and the _absolute_ path of the (temporarily) cached file.

Agreed, more or less:  we need to know the path within the results (which, presumably is and should be relative, but I don't really care so long as we understand that it is _within the results_), and we need to know the location of the cached file (which currently should be a file system path -- an absolute one is probably best -- but will someday be an S3 URI or a Redis identifier or something similar).

> So if the "actual path" for this cache map is going to be the file at the end of the link chain, we really do need to `readlink` and keep all our paths relative to the base of the tarball.

This is a good point; however, I think `resolve()` will do the job, so long as we vet the result.  If all the symlinks in the chain are relative references, then I believe that `resolve()` will come up with _an_ answer, and then we just have to ensure that it points within the result.  (And, I think it will be easier to use `resolve()` than to roll our own.)